src/base/bittorrent/tracker.cpp

```diff
@@ -130,19 +132,27 @@ Http::Response Tracker::processRequest(const Http::Request &request, const Http:
 
 void Tracker::respondToAnnounceRequest()
 {
-    const QStringMap &gets = m_request.gets;
+    QMap<QString, QByteArray> queryParams;
+    // Parse GET parameters
+    using namespace Utils::ByteArray;
+    for (const QByteArray &param : copyAsConst(splitToViews(m_request.query, "&"))) {
+        const int sepPos = param.indexOf('=');
+        if (sepPos >= 0) // ignores params without value
+            queryParams[midView(param, 0, sepPos)] = param.mid(sepPos + 1);
```

Comment:
`midView(param, sepPos + 1);`?

Reply:
queryParam key is QString. So QByteArray::mid() will produce unnecessary data copying.